backend/workflow_manager/workflow_v2/execution.py

```diff
@@ -305,6 +306,37 @@ def publish_final_workflow_logs(
             f"{successful_files} successfully executed and {failed_files} error(s)"
         )
 
+    def publish_average_cost_log(self, execution_id, total_files):
+
+        try:
+            total_cost = UsageHelper.get_aggregated_cost(execution_id)
+            average_cost = round(total_cost / total_files, 5)
+            self.publish_log(
+                message=(
+                    f"The average cost per file for execution '{execution_id}' "
+                    f"is '${average_cost:}'. Total cost: '${total_cost:}'"
+                )
+            )
+        except TypeError as e:
+            self.publish_log(
+                message=(
+                    f"Unable to calculate cost for execution '{execution_id}'. "
+                    f"Cost data may be unavailable or incomplete."
+                )
+            )
+
+            logger.error(
+                f"Error calculating cost for execution '{execution_id}': "
+                f"{str(e)}. Total cost: {total_cost}, Total files: {total_files}"
+            )
```

Comment:
```suggestion
            logger.error(
                f"Error calculating cost for execution '{execution_id}' of '{total_files}' files : "
                f"{str(e)}.\nContinuing execution"
            )
```
Ensure that this is within the line length if you accept this suggestion